compiler-plugin/src/main/kotlin/arrow/meta/internal/annotation/MetaAnnotations.kt

```diff
@@ -0,0 +1,7 @@
+package arrow
+
+@Retention(AnnotationRetention.SOURCE)
+@Target(AnnotationTarget.EXPRESSION, AnnotationTarget.CLASS, AnnotationTarget.TYPEALIAS,
+  AnnotationTarget.FIELD, AnnotationTarget.CONSTRUCTOR, AnnotationTarget.LOCAL_VARIABLE,
+  AnnotationTarget.FUNCTION, AnnotationTarget.TYPE_PARAMETER, AnnotationTarget.TYPE)
+internal annotation class synthetic
```

Comment:
Hi @i-walker , this annotation already exists in `arrow-annotations`: https://github.com/arrow-kt/arrow/blob/master/modules/meta/arrow-annotations/src/main/java/arrow/synthetic.kt

Reply:
I know
It fixed my issue with compilation, for the task. So the only difference is to turn `arrow-synthetic` to a `Source` Annotation, in Arrow it is a RUNTIME annotation. Which works perfectly in the test environment. But it doesn't in a project. 

I cant confidently say why exactly. But their might be also another solution to this issue.
When we want `arrow.synthetic` to stay a RunTime annotation.

Reply:
Hi mate! Thanks for the info!! I don't know the strategy so I'm talking without knowledge enough, however, my comment comes from this concern: maintaining duplicated annotations.

I updated `arrow-annotations` library a few days ago and that library is being used in all the Arrow Meta Compiler Plugin examples.

Besides, `synthetic` annotation from `arrow-annotations` library is located in `arrow` package so it's referred as:

```
arrow.synthetic
```

and this new annotation is located in `arrow.meta.internal.annotation` so it will be referred as 

```
arrow.meta.internal.annotation.synthetic
```

So it's just a warning and I hope it's useful for you!! :+1:

Reply:
nope. this annotation is still referenced as arrow.synthetic if you check the package definition.

Reply:
I dont mind closing this issue and moving it to arrow. At the time I opened the PR I thought there were to separate ones. And the other was from the legacy 'arrow-meta'